src/lib.rs

```diff
@@ -1,3 +1,25 @@
+//! A logger implementation to use RTT with the Rust `log` crate.
+//!
+//! ```
+//! use log::{info, LevelFilter};
+//! use rtt_logger::RTTLogger;
+//! use rtt_target::rtt_init_print;
+//!
+//! // logger configuration
+//! const LOG_LEVEL: LevelFilter = LevelFilter::Info;
+//! static LOGGER: RTTLogger = RTTLogger::new(LOG_LEVEL);
+//!
+//! fn main() {
+//!     // logger setup
+//!     rtt_init_print!();
+//!     log::set_logger(&LOGGER).unwrap();
+//!     log__set_max_level(LOG_LEVEL);
```

Comment:
```suggestion
//!     log::set_max_level(LOG_LEVEL);
```

I guess.

Otherwise, like inhttps://github.com/quartiq/stabilizer/blob/2498a3e5495bb0a48a96fa581f1ba645f9d6eba8/src/hardware/setup.rs#L248-L250

It would be great to get some github actions coverage of this with a minimal setup.

Reply:
Whoops, good catch. I've updated this.

CI would be nice, but there are some annoying issues with doctests and cross compiling - you can't run them with `cross` right now, and cross compiling is behind a feature flag. Probably not impossible to figure something out, but I'm not sure what that looks like. https://github.com/cross-rs/cross/issues/225

Reply:
Ack. But just clippy and fmt would also go some way for the entire repo.
But that's out of scope for the pr here.